src/org/jgroups/util/ThreadPool.java

```diff
@@ -171,10 +176,26 @@ public boolean execute(Runnable task) {
         catch(RejectedExecutionException ex) {
             tp.getMessageStats().incrNumRejectedMsgs(1);
             // https://issues.redhat.com/browse/JGRP-2403
+            String threadDump = Util.dumpThreads();
```

Comment:
This `org.jgroups.util.Util#dumpThreads` call is in the wrong place. It is expensive so make sure to call it at the code path you are certainly going to use the result.

Reply:
Ops. Thanks!